core/src/main/java/org/jivesoftware/sparkimpl/certificates/SparkTrustManager.java

```diff
@@ -189,24 +185,32 @@ private void validatePath(X509Certificate[] chain)
         PKIXBuilderParameters parameters = new PKIXBuilderParameters(trustStore, certSelector);
 
         if (acceptRevoked == false) {
-            if (checkCRL) {
-                // check add CRL store and download CRL list to this store.
+            // check OCSP, CRL serve as backup
+            if (checkOCSP && checkCRL) {
+                // OCSP checking is done according to Java PKI Programmer's Guide, PKIXRevocationChecker was added in Java 8:
+                // https://docs.oracle.com/javase/8/docs/technotes/guides/security/certpath/CertPathProgGuide.html#PKIXRevocationChecker
+                PKIXRevocationChecker checker = (PKIXRevocationChecker) certPathBuilder.getRevocationChecker();
+                // soft fail option mean that OCSP or CRL must pass validation, in case when OCSP cannot be
+                // validated it will validate CRL and then if both cannot be checked it will fail
+                checker.setOptions(EnumSet.of(PKIXRevocationChecker.Option.SOFT_FAIL));
```

Comment:
Should this be configurable? Perhaps the end-user should control if SOFT_FAIL should apply.